CONTRIBUTING.md

```diff
@@ -1,252 +1,63 @@
+### Submitting pull requests
 
-- Create the fork on github, using the fork button.
-- Go to the original repo checked out locally (i.e. `$GOPATH/src/github.com/cosmos/ibc-go`)
-- `git remote add fork git@github.com:colin-axner/ibc-go.git`
+Unless you feel confident your change will be accepted (trivial bug fixes, code cleanup, etc) you should first create an issue to discuss your change with us. This lets us all discuss the design and proposed implementation of your change, which helps ensure your time is well spent and that your contribution will be accepted.
 
-Now `fork` refers to my fork and `origin` refers to the ibc-go version.
-So I can `git push -u fork main` to update my fork, and make pull requests to ibc-go from there.
-Of course, replace `colin-axner` with your git handle.
+Looking for a good place to start contributing? The issue tracker is always the first place to go. Issues are triaged to categorize them:
 
-To pull in updates from the origin repo, run
+- Check out some [`good first issue`s](https://github.com/cosmos/ibc-go/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22). These are issues whose scope of work should be pretty clearly specified and they are best suited for developers new to ibc-go (i.e. no deep knowledge of Cosmos SDK or ibc-go is required). For example, some of these issues may involve improving the logging, emitting new events or removing unsused code.
+- Or pick up a [`help wanted`](https://github.com/cosmos/ibc-go/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) issue. These issues should be a bit more involved than the good first issues and the developer working on them would benefit from some familiarity already with the codebase. This type of issues may involve adding new (or extending the functionality of existing) gRPC endpoints, bumping the version of Cosmos SDK or Tendermint or fixing bugs.
 
-- `git fetch origin`
-- `git rebase origin/main` (or whatever branch you want)
+If you would like to contribute, follow this process:
 
-Please don't make Pull Requests from `main`.
+1. If the issue is a proposal, ensure that the proposal has been accepted.
+2. Ensure that nobody else has already begun working on this issue. If they have, make sure to contact them to collaborate.
+3. If nobody has been assigned for the issue and you would like to work on it, comment on the issue to inform the community of your intentions to begin work. Then we will be able to assign the issue to you, making it visible for others that this issue is being tackled. If you end up not creating a pull request for this issue, please comment on the issue as well, so that it can be assigned to somebody else.
+4. Follow standard GitHub best practices: fork the repo, branch from the HEAD of `main`, make some commits, and submit a PR to `main`. For core developers working within the ibc-go repo, branches must be named with the convention `{moniker}/{issue#}-branch-name` to ensure a clear ownership of branches.
```

Comment:
External contributors should allow maintainers to edit their branch

Reply:
I mention that [here](https://github.com/cosmos/ibc-go/blob/b55836829eda64082dba8b69c24276e90df72228/docs/dev/pull-requests.md#pull-request-guidelines), but I can repeat it here as well.